test/Microsoft.TestPlatform.Common.UnitTests/Utilities/RunSettingsUtilitiesTests.cs

```diff
@@ -34,11 +35,11 @@ public void CreateRunSettingsShouldThrowExceptionWhenInvalidXmlStringIsPassed()
         [TestMethod]
         public void CreateRunSettingsShouldReturnValidRunSettings()
         {
-            TestPluginCacheTests.SetupMockExtensions();
+            TestPluginCacheHelper.SetupMockExtensions(typeof(RunSettingsUtilitiesTests));
             string runsettings = @"<RunSettings><RunConfiguration><ResultsDirectory>.\TestResults</ResultsDirectory></RunConfiguration ><DummyMSTest><FORCEDLEGACYMODE>true</FORCEDLEGACYMODE></DummyMSTest></RunSettings>";
             var result= RunSettingsUtilities.CreateAndInitializeRunSettings(runsettings);
             Assert.AreEqual("<DummyMSTest><FORCEDLEGACYMODE>true</FORCEDLEGACYMODE></DummyMSTest>", DummyMsTestSetingsProvider.StringToVerify);
```

Comment:
If this assert fails, `TestPluginCacheHelper.ResetExtensionsCache();` wont run. Is that expected? Should we use `try..finally`?

Reply:
It worked like that till now. So I would not change it now.